packages/react-admin/src/Selected.tsx

```diff
@@ -21,41 +21,45 @@ const ProgressContainer = styled.div`
     justify-content: center;
 `;
 
+const SelectEdit = (props: IProps) => {
+    console.log(props.query);
+    const queryResult = useQuery(props.query!, { variables: { id: props.selectedId } });
+    if (queryResult.error) {
+        const ErrorComponent = props.components?.error;
+        return ErrorComponent !== undefined ? (
+            <ErrorComponent error={queryResult.error} />
+        ) : (
+            <Card>
+                <Box padding={4}>
+                    <Typography>Error :( {queryResult.error.toString()}</Typography>
+                </Box>
+            </Card>
+        );
+    }
+    if (queryResult.loading || !queryResult.data) {
+        return (
+            <ProgressContainer>
+                <CircularProgress />
+            </ProgressContainer>
+        );
+    }
+    if (!props.dataAccessor) {
+        throw new Error("dataChild prop is required");
+    }
+    return <>{props.children(queryResult.data[props.dataAccessor], { selectionMode: "edit" })}</>;
+};
+
 export function Selected(props: IProps) {
-    const ErrorComponent = props.components?.error;
     let row;
     if (props.rows) {
         row = props.rows.find((i) => String(i.id) === String(props.selectedId)); // compare as strings as selectedId might come from url
     }
 
-    const queryResult = useQuery(props.query!, { variables: { id: props.selectedId }, skip: props.query === undefined });
```

Comment:
The query is not executed conditionally. I think the the problem was that an if-condition is defined (line 27) before a hook is defined (line 31). So swapping the 2 statements should fix any hook-lint warnings.

Or do you mean with "executed conditionally" the passed "skip"-option?

Generally a good way to handle conditional queries with apollo is using `useLazyQuery` instead of `useQuery`.
https://www.apollographql.com/docs/react/data/queries/#executing-queries-manually

Reply:
With "conditionally" I meant the skip statement. useQuery does not accept an empty query, even though the query would be skipped due to the skip-property. Maybe the change of the behavior was introduced with the Apollo-update, I didn't double-check this though.